controllers/authorino_controller.go

```diff
@@ -0,0 +1,619 @@
+	envVar := []k8score.EnvVar{}
+
+	if !authorino.Spec.ClusterWide {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  "WATCH_NAMESPACE",
+			Value: authorino.GetNamespace(),
+		})
+	}
+
+	if authorino.Spec.AuthConfigLabelSelectors != "" {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.AuthConfigLabelSelector,
+			Value: fmt.Sprint(authorino.Spec.AuthConfigLabelSelectors),
+		})
+	}
+
+	if authorino.Spec.SecretLabelSelectors != "" {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.SecretLabelSelector,
+			Value: fmt.Sprint(authorino.Spec.SecretLabelSelectors),
+		})
+	}
+
+	// external auth service via GRPC
+	if authorino.Spec.Listener.Port != nil {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.ExtAuthGRPCPort,
+			Value: fmt.Sprint(authorino.Spec.Listener.Port),
+		})
+	}
+	if authorino.Spec.Listener.CertPath != "" {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.TSLCertPath,
+			Value: authorino.Spec.Listener.CertPath,
+		})
+	}
+	if authorino.Spec.Listener.CertKeyPath != "" {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.TSLCertKeyPath,
+			Value: authorino.Spec.Listener.CertKeyPath,
+		})
+	}
+
+	// OIDC service
+	if authorino.Spec.OIDCServer.Port != nil {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.OIDCHTTPPort,
+			Value: fmt.Sprint(authorino.Spec.OIDCServer.Port),
+		})
+	}
+	if authorino.Spec.OIDCServer.CertKeyPath != "" {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.OIDCTSLCertPath,
+			Value: authorino.Spec.OIDCServer.CertPath,
+		})
+	}
+	if authorino.Spec.OIDCServer.CertKeyPath != "" {
+		envVar = append(envVar, k8score.EnvVar{
+			Name:  api.OIDCTLSCertKeyPath,
+			Value: authorino.Spec.OIDCServer.CertKeyPath,
+		})
+	}
+
+	return envVar
+}
+
+func (r *AuthorinoReconciler) authorinoDeploymentChanges(existingDeployment, desiredDeployment *k8sapps.Deployment) bool {
+	changed := false
+
+	if len(desiredDeployment.Spec.Template.Spec.Containers) != 1 {
+		// error
+	}
+
+	existingContainer := existingDeployment.Spec.Template.Spec.Containers[0]
+	desiredContainer := desiredDeployment.Spec.Template.Spec.Containers[0]
+
+	if existingContainer.Image != desiredContainer.Image {
+		changed = true
+	}
+
+	if existingContainer.ImagePullPolicy != desiredContainer.ImagePullPolicy {
+		changed = true
+	}
+
+	// checking envvars
+	existingEnvvars := existingContainer.Env
+	desiredEnvvars := desiredContainer.Env
+	for envIndex, existingEnvvar := range existingEnvvars {
+		desiredEnvvar := desiredEnvvars[envIndex]
+		if existingEnvvar.Name == desiredEnvvar.Name && existingEnvvar.Value != desiredEnvvar.Value {
+			changed = true
+		}
+	}
+	return changed
+}
+
+func (r *AuthorinoReconciler) createAuthorinoPermission(authorino *api.Authorino, operatorNamespace string) error {
+	prefix := authorino.GetName()
+	authorinoInstanceNamespace := authorino.GetNamespace()
+	authorinoClusterRoleName := "authorino-operator-authorino-manager-role"
```

Comment:
To make it work while trying locally (following the instructions in the description of the PR), I had to change this to:

```suggestion
	authorinoClusterRoleName := "authorino-manager-role"
```